### PR TITLE
GWH 2022 Module

### PR DESCRIPTION
**Pull request by hymccord**
Testing out moving ajaxSuccess handlers from main.js to their own little files/components.  
Everything to do with golems has been moved to `golems.ts`. Url matching, parsing the response, sending to google script, and showing result banner.

Each "handler" as two methods: `match` and `execute`.  
`match`: you are passed the xhr URL and you return a boolean if your handler can act on it (GWH also checks current time)
`execute`: do what you will with the xhr.responseJSON!

`golem.types.ts` contains all the type definitions
`golem.ts` is the implementation of the handler

The two external dependencies needed for golem submission are constructor injected. This is logging service and the showFlashMessage method to show the success/error banner.

### CODE REVIEW

**Comment by hymccord** [comment]
Folder structure is pretty flexible.

Maybe `./scripts/components/ajaxhandlers/golem.ts`? Because different folders/items could eventually go in components (stages, details).

File: src/scripts/ajax-handlers/golem.types.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+export type Rarity = "area" | "hat" | "scarf";
+
+/**
+ * The data that will be recorded externally
+ */
+export interface GolemPayload {
+    timestamp: number;
+    location: string;
+    loot: (GolemItem & {
+        rarity: Rarity
```
Comment (hymccord) [inline src/scripts/ajax-handlers/golem.types.ts:42]: Most of the typing is very straight forward. This is the most advanced portion that I learned about for this PR.

It's called an Instersection Type and it allows me to tack on the property without having to define and extra `GolemItem` type  that has the `rarity` property.
https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#intersection-types

File: jest.config.js [hunk 2]
```diff
@@ -13,4 +13,5 @@ module.exports = {
     moduleNameMapper: pathsToModuleNameMapper(compilerOptions?.paths || {}, {
         prefix: "<rootDir>/",
     }),
+    testEnvironment: 'jsdom',
```
Comment (hymccord) [inline jest.config.js:16]: The default environment for test is `node`. Changing it back to `jsdom` allows tests to interact with `document`, `window`, etc. Things you get in the browser.

**Comment by hymccord** [comment]
Updated ajax url to `winter_hunt_region.php` and response rewards will be user `responseJSON.golem_rewards`

**Review by AardWolf — APPROVED** [review]

**Review by AardWolf — DISMISSED** [review]

**Review by AardWolf — APPROVED** [review]

File: src/scripts/background.js [hunk 3]
```diff
@@ -187,11 +187,6 @@ chrome.runtime.onMessage.addListener((msg, sender, sendResponse) => {
         submitCrowns(msg.crowns).then(sendResponse);
         return true;
     }
-
-    if (msg.mhct_golem_submit === 1) {
-        submitGolems(msg.golems).then(sendResponse);
-        return true;
-    }
     // TODO: Handle other extension messages.
```
Comment (tehhowch) [inline src/scripts/background.js:190]: Note that moving this into the main script is why other extensions (like Privacy Badger) can block the request.

**Pull request merged**